tests/unit/middleware/test_geolocation.py

```diff
@@ -163,3 +163,10 @@ async def test_geolocation_invalid_scope_type(
 
     assert ScopeKey.GEOLOCATION not in scope
     assert len(caplog.messages) == 0
+
+
+def test_normalize_string() -> None:
+    """Test the normalization of strings with special characters"""
+    assert normalize_string("Kīhei") == "Kihei"
+    assert normalize_string("‘Aiea") == "Aiea"
+    assert normalize_string("Querétaro") == "Queretaro"
```

Comment:
suggestion to make this data driven (Note needs to be tested)

```suggestion
@pytest.mark.parametrize("input_string, expected_output", [
    ("Kīhei", "Kihei"),
    ("‘Aiea", "Aiea"),
    ("Querétaro", "Queretaro"),
])
def test_normalize_string(input_string: str, expected_output: str) -> None:
    """Test the normalization of strings with special characters"""
    assert normalize_string(input_string) == expected_output
```